sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusScope.cs

```diff
@@ -0,0 +1,627 @@
+        ///   of the scope; disposal removes the resource from Azure.
+        /// </summary>
+        ///
+        /// <seealso cref="IAsyncDisposable" />
+        ///
+        public sealed class QueueScope : IAsyncDisposable
+        {
+            /// <summary>Serves as a sentinel flag to denote when the instance has been disposed.</summary>
+            private bool _disposed = false;
+
+            /// <summary>
+            ///   The name of the Service Bus namespace associated with the queue.
+            /// </summary>
+            ///
+            public string NamespaceName { get; }
+
+            /// <summary>
+            ///  The name of the queue.
+            /// </summary>
+            ///
+            public string QueueName { get; }
+
+            /// <summary>
+            ///   A flag indicating if the queue should be removed when the scope is complete.
+            /// </summary>
+            ///
+            /// <value><c>true</c> if the queue was should be removed at scope completion; otherwise, <c>false</c>.</value>
+            ///
+            private bool ShouldRemoveAtScopeCompletion { get; }
+
+            /// <summary>
+            ///   Initializes a new instance of the <see cref="QueueScope"/> class.
+            /// </summary>
+            ///
+            /// <param name="serviceBusNamespaceName">The name of the Service Bus namespace to which the queue is associated.</param>
+            /// <param name="queueName">The name of the queue.</param>
+            /// <param name="shouldRemoveAtScopeCompletion">A flag indicating whether the queue should be removed when the scope is complete.</param>
+            ///
+            public QueueScope(string serviceBusNamespaceName,
+                              string queueName,
+                              bool shouldRemoveAtScopeCompletion)
+            {
+                NamespaceName = serviceBusNamespaceName;
+                QueueName = queueName;
+                ShouldRemoveAtScopeCompletion = shouldRemoveAtScopeCompletion;
+            }
+
+            /// <summary>
+            ///   Performs the tasks needed to remove the dynamically created Service Bus Queue.
+            /// </summary>
+            ///
+            public async ValueTask DisposeAsync()
+            {
+                if (!ShouldRemoveAtScopeCompletion)
+                {
+                    _disposed = true;
+                }
+
+                if (_disposed)
+                {
+                    return;
+                }
+
+                try
+                {
+                    var subscription = TestEnvironment.ServiceBusSubscription;
+                    var resourceGroup = TestEnvironment.ServiceBusResourceGroup;
+                    var token = await AquireManagementTokenAsync().ConfigureAwait(false);
+
+                    using var client = new ServiceBusManagementClient(new TokenCredentials(token)) { SubscriptionId = subscription };
```

Comment:
Subscriptions shouldn't apply for queues.

Reply:
The Azure subscription in which the resource group is based sure does.   😄

Reply:
> The Azure subscription in which the resource group is based sure does. 😄

Oh that kind of subscription..

Reply:
Apologies for the confusion.  Tweaked to be "Azure Subscription" to avoid conflating with the Service Bus connotation.